packages/wallet-apis/src/client.ts

```diff
@@ -1,17 +1,25 @@
 import {
   createWalletClient,
+  type Address,
   type Chain,
   type JsonRpcAccount,
   type LocalAccount,
 } from "viem";
 import { smartWalletActions } from "./decorators/smartWalletActions.js";
 import type { AlchemyTransport } from "@alchemy/common";
 
-export type CreateSmartWalletClientParams = {
+export type CreateSmartWalletClientParams<
+  TAccount extends Address | undefined = Address | undefined,
+> = {
   account: LocalAccount | JsonRpcAccount;
   transport: AlchemyTransport;
   chain: Chain;
   policyIds?: string[];
+  // TODO(jh): unsure how i feel about this, but it gets us the typesafety of not having to specify a `from` address on every call.
+  // do the ergonomics of this feel better or worse than having to pass the `signer` or `owner` separately from the `account`?
+  // or should we not care about typesafety of including the `from` address and just try to use the account that's active
+  // in the internal cache & throw an error if there's no account or from address?
```

Comment:
i think i'm leaning towards the latter if we are happy w/ the `account` being the signer.

still thinking about that a bit though & curious to hear what others think.